### PR TITLE
feat: add initiated by in history and rollback view

### LINKED ISSUE #16612: Add user info in Argo CD history UI
# Summary

In Argo CD UI, it would be useful to have in the history panel the information of the user who triggered the sync.

# Motivation

Better visibility of who/what is triggering Argo CD Syncs

# Proposal

Identify the possibility to add additional information in the history panel in Argo CD UI of the user that triggered the sync.
See image:
image

### PR DESCRIPTION
**Pull request by robinlieb**
Closes #16612

As mentioned in the issue this PR adds **Initiated by** in the history and rollback view.

argocd-16612

Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [X] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [X] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [X] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [X] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [ ] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

File: ui/src/app/applications/components/application-deployment-history/application-deployment-history.tsx
Comment (keithchong) [inline ui/src/app/applications/components/application-deployment-history/application-deployment-history.tsx:46]: Although this is fine to me, maybe confirm with Leo if he wanted this to be on a separate line or on the same line as "Time to deploy" as depicted by his screenshot in the issue?

**Review by leoluz — APPROVED** [review]
Thank you @robinlieb for working on this!
Will wait for @keithchong to conclude the UI review before merging.

**Review by keithchong — APPROVED** [review]
@leoluz , LGTM

**Review by leoluz — CHANGES_REQUESTED** [review]
@robinlieb There is one last request in this PR. As you are changing the Application CRD, this needs to be stated in the "upgrading" doc.

As we just cut the 2.10 branch a few days ago, the file is not created yet. It needs to be created in:
`/docs/operator-manual/upgrading/2.10-2.11.md` and linked here:
https://github.com/argoproj/argo-cd/blob/7847e7f393c0f83a12d4733a3a887f0875dbb086/mkdocs.yml#L131

See this doc as an example:
https://argo-cd.readthedocs.io/en/stable/operator-manual/upgrading/1.6-1.7/#appproject-tokens-moved-to-status-field

**Comment by robinlieb** [comment]
@leoluz thanks for the hint! Added the respective upgrading doc and linked it in the `mkdocs.yml`

**Review by leoluz — APPROVED** [review]
LGTM and thank you!

**Comment by ericblackburn** [comment]
When we manually sync, it shows initiated by "Unknown" instead of using my Username that shows up on the /user-info page.  I don't think this is working with progressive-sync either.

**Comment by eliya-gloat** [comment]
Same for me with release - v2.11.7+e4a0246 - All actions appears as user "unknown" for field "Initiated By"
v2.11.7+e4a0246

**Pull request merged**
